### PR TITLE
Condition service directory parameter to Accommodate core ci

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Allow actual directory name to be passed in core/ci.yml but still support the functionality of building all services.

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 1]
```diff
@@ -7,6 +7,10 @@ stages:
     jobs:
     - template: ../jobs/archetype-sdk-client.yml
       parameters:
+        ${{ if eq(parameters['ServiceDirectory'], 'core') }}:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml]: Can you make ServicetoBuild a parameter instead? And set it to `*` in core and default it to ServiceDirectory everywhere else? I'd prefer not to special case core in the template directly.

Comment (chidozieononiwu) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml]: Ok, I can do that, but that'll will mean modifying all the `ci.yml` files which I assumed will not be such a good idea.

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-sdk-client.yml]: You should only need to modify the core ci.yml file. The others will get defaulted in this template file.

File: eng/pipelines/templates/stages/archetype-sdk-client.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ stages:
     jobs:
     - template: ../jobs/archetype-sdk-client.yml
       parameters:
+        ServiceToBuild: ${{parameters.ServiceDirectory}}
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-client.yml]: I don't think this is going to do what you want. I think you want something like:

```
 ServiceToBuild: ${{ coalesce(parameters.ServiceToBuild, parameters.ServiceDirectory) }}
```
see more information about coalesce at https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#coalesce

**Review by weshaggard — APPROVED** [review]

**Pull request merged**